ui/src/forms/editors/index.ts

```diff
@@ -73,16 +74,94 @@ export const enumSelect: Lazy<EnumSelectEditor> = {
   }
 }
 
+declare module '@hydrofoil/shaperone-core/components' {
+  interface InstancesSelectEditor {
+    hasFreeTextQueryVariable(params: SingleEditorRenderParams<InstancesSelect>): boolean
+  }
+}
+
 export const autoComplete: Lazy<InstancesSelectEditor> = {
   ...instancesSelectCore,
   editor: dash.AutoCompleteEditor,
+  hasFreeTextQueryVariable (params) {
+    return !!this.searchTemplate?.(params)?.mapping
+      .some(({ property }) => property?.equals(hydra.freetextQuery))
+  },
+  init (params) {
+    if (!this.hasFreeTextQueryVariable(params)) {
+      return instancesSelectCore.init?.call(this, params) || true
+    }
+
+    const { form, property, value, updateComponentState } = params
```

Comment:
The "overloaded" `init` function is used when collection is filtered on input so that the initial value gets dereferenced and its label used as the placeholder.